Hygiene related changes to drop in support page (#25699)
DataDog/documentation: content/en/opentelemetry/interoperability/instrumentation_libraries.md

  tags: ['otel instrumentation']
---

> Datadog's support for OpenTelemetry instrumentations allows you to leverage community-built OpenTelemetry-compatible instrumentations. You can extend Datadog's tracing capabilities to these frameworks, ensuring no part of your tech stack is left unmonitored.

Reviewer: Can we switch "leverage" to "use"? Also comes across buzzy

Author: Done

> Datadog's support for OpenTelemetry instrumentations allows you to leverage community-built OpenTelemetry-compatible instrumentations. You can extend Datadog's tracing capabilities to these frameworks, ensuring no part of your tech stack is left unmonitored.

Author: I fixed it. Removed tge buzzy stuff.

> Datadog’s support for OpenTelemetry instrumentations now allows you to leverage community-built OpenTelemetry-compatible instrumentations. You can now extend Datadog’s tracing capabilities to these frameworks, ensuring no part of your tech stack is left unmonitored.

Reviewer 2: From the automated suggestions:
- Use straight quotes instead of smart quotes.
- Avoid temporal words like 'now'.

```suggestion
Datadog's support for OpenTelemetry instrumentations let you to leverage community-built OpenTelemetry-compatible instrumentations. You can extend Datadog's tracing capabilities to these frameworks, ensuring no part of your tech stack is left unmonitored.
```

Author: Addressed

> Datadog's support for OpenTelemetry instrumentations allows you to use OpenTelemetry-compatible instrumentations. You can extend Datadog's tracing capabilities to these frameworks. This provides observability for libraries not covered by Datadog SDKs without changing SDKs.

Reviewer 3: ```suggestion
Datadog supports OpenTelemetry-compatible instrumentations which provides observability for libraries not covered by Datadog SDKs without changing SDKs. You can extend Datadog's tracing capabilities to these frameworks.
```
Maybe something like this is more clear?

## Configuration

> You can also configure Datadog SDKs by setting the same [environment variables supported by OpenTelemetry][16].

Reviewer 3: ```suggestion
You can configure Datadog SDKs by setting the same [environment variables supported by OpenTelemetry][16].
```
What's the also referencing?

## Configuration

> Datadog SDKs also support configuring [OpenTelemetry environment variables][16].

Reviewer 2: I'm not sure about this new phrasing. The SDKs don't "support configuring OpenTelemetry environment variables." Users configure the tracer by setting environment variables.

Maybe reword this to something like:
> You can also configure Datadog SDKs by setting the same environment variables supported by OpenTelemetry.

or

> The Datadog SDKs support many of the same environment variables supported by OpenTelemetry.

Author: Yea. Making this change

> You can also configure Datadog SDKs by setting the same environment variables supported by [OpenTelemetry][16].

Author: Done

You can also configure Datadog SDKs by setting the same environment variables supported by [OpenTelemetry][16].

## Language support

Datadog SDKs implement the OpenTelemetry API by overriding the default implementations in the OpenTelemetry SDK. However, note the following limitations:

- Operations only supported by the OpenTelemetry SDK are not supported (for example, SpanProcessors or OTLP Trace Exporters).

> - Datadog SDKs do not support OpenTelemetry Metrics and Logs APIs. To use OpenTelemetry Logs and Metrics APIs, use [OTLP Ingest][12].

Author: Added

2. Copy the OpenTelemetry extension JAR containing the instrumentation to the same container as the application.
3. Set the `otel.javaagent.extensions` system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable to the extension JAR path.

## Example

> Here’s a step-by-step example using R2DBC in Java to illustrate how you can use OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. 

Reviewer 4: ```suggestion
Here's a step-by-step example using R2DBC in Java to illustrate how you can use OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. 
```

```sh
curl -Lo opentelemetry-javaagent-r2dbc.jar \
  'https://repo1.maven.org/maven2/io/opentelemetry/javaagent/instrumentation/opentelemetry-javaagent-r2dbc-1.0/2.5.0-alpha/opentelemetry-javaagent-r2dbc-1.0-2.5.0-alpha.jar'

mvn spring-boot:run -Dstart-class=com.baeldung.pagination.PaginationApplication \
  -Dspring-boot.run.jvmArguments='-javaagent:dd-java-agent.jar -Ddd.trace.otel.enabled=true -Dotel.javaagent.extensions=opentelemetry-javaagent-r2dbc.jar -Ddd.trace.split-by-tags=db.name,db.sql.table -Ddd.trace.debug=true'
```

Open `http://127.0.0.1:8080/products` to exercise the product query. With this setup, you are using OpenTelemetry's instrumentation to ensure full observability for R2DBC queries.

<div class="alert alert-warning">
Versions 2.6.0-alpha and later of these OpenTelemetry instrumentations are not supported by the Datadog Java SDK.
</div>

## Verified OpenTelemetry extensions

| Framework           | Versions | OpenTelemetry Extension                         | Instrumentation Names |
|---------------------|----------|-------------------------------------------------|-----------------------|

> | Apache CXF (Jax-WS) | 3.0+     | [opentelemetry-javaagent-jaxws-2.0-cxf-3.0][10] | `otel.cxf`                 |

Author: Yep and rightly so.

2. Copy the OpenTelemetry extension JAR containing the instrumentation to the same container as the application.
3. Set the `otel.javaagent.extensions` system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable to the extension JAR path.

> Here’s a step-by-step example using R2DBC in Java to illustrate how you can drop OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. R2DBC is the reactive version of JDBC, commonly used for reactive database interactions in Java applications. Datadog’s Java agent does not provide out-of-the-box instrumentation for R2DBC queries. 

Reviewer 4: ```suggestion
Here's a step-by-step example using R2DBC in Java to demonstrate integrating OpenTelemetry instrumentation into your service to send data to Datadog and capture all the missing spans. R2DBC is the reactive version of JDBC, commonly used for reactive database interactions in Java applications. Datadog's Java agent does not provide out-of-the-box instrumentation for R2DBC queries. 
```

Author: Addressed

> Here’s a step-by-step example using R2DBC in Java to illustrate how you can drop OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. R2DBC is the reactive version of JDBC, commonly used for reactive database interactions in Java applications. Datadog’s Java agent does not provide out-of-the-box instrumentation for R2DBC queries. 

Reviewer 4: If you want some separation from Setup, suggest adding a separate header.

Author: Yep! Moved it to an example section

> Here’s a step-by-step example using R2DBC in Java to illustrate how you can drop OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. R2DBC is the reactive version of JDBC, commonly used for reactive database interactions in Java applications. Datadog’s Java agent does not provide out-of-the-box instrumentation for R2DBC queries. 

Reviewer 2: > Use straight quotes instead of smart quotes.

```suggestion
Here's a step-by-step example using R2DBC in Java. It illustrates how you can drop OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. R2DBC is the reactive version of JDBC, commonly used for reactive database interactions in Java applications. Datadog's Java agent does not provide out-of-the-box instrumentation for R2DBC queries. 
```

> Download the OpenTelemetry R2DBC agent and run your Spring Boot application with both the Datadog Java agent and the OpenTelemetry R2DBC agent.

Reviewer 4: ```suggestion
1. Download the OpenTelemetry R2DBC agent and run your Spring Boot application with both the Datadog Java agent and the OpenTelemetry R2DBC agent.
```

```sh
git clone https://github.com/eugenp/tutorials
cd tutorials/spring-reactive-modules/spring-reactive-data

curl -Lo dd-java-agent.jar 'https://dtdg.co/latest-java-tracer'

curl -Lo opentelemetry-javaagent-r2dbc.jar \
  'https://repo1.maven.org/maven2/io/opentelemetry/javaagent/instrumentation/opentelemetry-javaagent-r2dbc-1.0/2.5.0-alpha/opentelemetry-javaagent-r2dbc-1.0-2.5.0-alpha.jar'

mvn spring-boot:run -Dstart-class=com.baeldung.pagination.PaginationApplication \
  -Dspring-boot.run.jvmArguments='-javaagent:dd-java-agent.jar -Ddd.trace.otel.enabled=true -Dotel.javaagent.extensions=opentelemetry-javaagent-r2dbc.jar -Ddd.trace.split-by-tags=db.name,db.sql.table -Ddd.trace.debug=true'
```

> Open `http://127.0.0.1:8080/products` to exercise the product query. With this setup, you’re now using OpenTelemetry’s instrumentation to ensure full observability for R2DBC queries.

Reviewer 4: ```suggestion
2. Open `http://127.0.0.1:8080/products` to exercise the product query. With this setup, you're using OpenTelemetry’s instrumentation to ensure full observability for R2DBC queries.
```

Author: Addressed

> Open `http://127.0.0.1:8080/products` to exercise the product query. With this setup, you’re now using OpenTelemetry’s instrumentation to ensure full observability for R2DBC queries.

Reviewer 2: ```suggestion
Open `http://127.0.0.1:8080/products` to exercise the product query. With this setup, you are using OpenTelemetry's instrumentation to ensure full observability for R2DBC queries.
```

<div class="alert alert-warning">

> Additionally, note that the `-Ddd.trace.split-by-tags=db.name,db.sql.table` setting gives R2DBC spans a service name based on the table. If you don't want R2DBC spans to appear under a separate service then that setting can be dropped.

Reviewer 4: What do you mean by drop? Would remove or omit mean the same thing?
```suggestion
The `-Ddd.trace.split-by-tags=db.name,db.sql.table` setting gives R2DBC spans a service name based on the table. If you don't want R2DBC spans to appear under a separate service then drop that setting.
```

</div>

> <div class="alert alert-warning">

Reviewer 4: Suggest moving this alert to ## Compatibility requirements.

Author: Done!

2. Copy the OpenTelemetry extension JAR containing the instrumentation to the same container as the application.
3. Set the `otel.javaagent.extensions` system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable to the extension JAR path.

## Example

> Here's a step-by-step example using R2DBC in Java to illustrate how you can use OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. 

Reviewer 3: ```suggestion
Here's a step-by-step example using R2DBC in Java to illustrate how you can add OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. 
```
